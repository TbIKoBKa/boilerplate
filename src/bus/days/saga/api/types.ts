// Types
import { Days } from '../../types';

export type FetchDaysContract = () => Promise<Days>
