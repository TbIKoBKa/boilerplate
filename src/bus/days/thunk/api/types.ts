// Types
import { Days } from '../../types';

export type FetchDays = () => Promise<Days>
