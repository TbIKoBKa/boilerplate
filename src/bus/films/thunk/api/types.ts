// Types
import { Films } from '../../types';

export type FetchFilms = () => Promise<Films>
