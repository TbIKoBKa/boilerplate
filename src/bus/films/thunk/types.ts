// Fetch
export const FETCH_FILMS_ASYNC = 'FETCH_FILMS_ASYNC';
export type fetchFilmsActionAsync = {
    type: typeof FETCH_FILMS_ASYNC
}
export type FetchFilmsContract = () => fetchFilmsActionAsync
