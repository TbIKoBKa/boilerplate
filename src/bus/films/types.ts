// Core

export type Film = {
    title: string,
    episode_id: number,
    opening_crawl: string,
    director: string,
    producer: string,
    release_date: string,
    characters: Array<string>,
    planets: Array<string>,
    starships: Array<string>,
    vehicles: Array<string>,
    species: Array<string>,
    created: string,
    edited: string,
    url: string,
    [key: string]: any,
}

export type Films = Array<Film>;

// Set
export const SET_FILMS = 'SET_FILMS';
export type SetFilmsActionType = {
    type: typeof SET_FILMS,
    payload: Films,
}
export type SetFilmsContract = (payload: Films) => SetFilmsActionType

export type PeopleActionTypes =
    | SetFilmsActionType
