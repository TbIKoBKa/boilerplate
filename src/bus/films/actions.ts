// Types
import * as types from './types';

export const setFilmsAction: types.SetFilmsContract = (payload) => ({
    type: types.SET_FILMS,
    payload,
});
