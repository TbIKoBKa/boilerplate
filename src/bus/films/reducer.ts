// Core
import { Reducer } from 'redux';

// Types
import * as types from './types';

const inittialState: types.Films = [];

export const filmsReducer: Reducer<types.Films, types.PeopleActionTypes> = (state = inittialState, action) => {
    switch (action.type) {
        case types.SET_FILMS:
            return action.payload;
        default:
            return state;
    }
};
