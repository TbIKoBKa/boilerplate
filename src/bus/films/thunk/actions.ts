import * as types from './types';

export const fetchFilmsActionAsync: types.FetchFilmsContract = () => ({
    type: types.FETCH_FILMS_ASYNC,
});
