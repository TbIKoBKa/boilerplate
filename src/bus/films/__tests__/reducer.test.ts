// Reducer
import { filmsReducer as films } from '../reducer';

// Actions
import { setFilmsAction } from '../actions';

// Types
import { Films } from '../types';

const testFilms: Films = [
    {
        title:         'Naruto',
        episode_id:    0,
        characters:    [ 'Naruto' ],
        created:       '2021',
        director:      'Naruto Uzumaki',
        edited:        '2021',
        opening_crawl: 'Best film',
        planets:       [ 'Kanoha' ],
        producer:      'Saske Uchiha',
        release_date:  '2021',
        species:       [ 'Something' ],
        starships:     [ 'Starship' ],
        url:           'naruto.watch.tv',
        vehicles:      [ 'Foots', 'Hands' ],
    },
];

describe('planets reducer', () => {
    test('should handle set action', () => {
        expect(films(void 0, setFilmsAction(testFilms))).toMatchSnapshot();
    });
});
