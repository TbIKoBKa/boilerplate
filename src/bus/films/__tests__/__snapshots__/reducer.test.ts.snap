// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`planets reducer should handle set action 1`] = `
Array [
  Object {
    "characters": Array [
      "Naruto",
    ],
    "created": "2021",
    "director": "Naruto Uzumaki",
    "edited": "2021",
    "episode_id": 0,
    "opening_crawl": "Best film",
    "planets": Array [
      "Kanoha",
    ],
    "producer": "Saske Uchiha",
    "release_date": "2021",
    "species": Array [
      "Something",
    ],
    "starships": Array [
      "Starship",
    ],
    "title": "Naruto",
    "url": "naruto.watch.tv",
    "vehicles": Array [
      "Foots",
      "Hands",
    ],
  },
]
`;
